lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -832,5 +866,23 @@ public BulkScorer bulkScorer(LeafReaderContext context) throws IOException {
       return new DefaultBulkScorer(new ConstantScoreScorer(this, 0f, ScoreMode.COMPLETE_NO_SCORES, disi));
     }
 
+    // Perform a cache load asynchronously
+    // @return true if asynchronous caching succeeded, false otherwise
+    private boolean cacheAsynchronously(LeafReaderContext context, IndexReader.CacheHelper cacheHelper) {
+      FutureTask<Void> task = new FutureTask<>(() -> {
+        DocIdSet localDocIdSet = cache(context);
```

Comment:
Since we are caching asynchronously, we might me hitting a closed segment. We should use `LeafReader#tryIncRef` to make sure the reader is still open and add tests for that case.